wandb/sdk/launch/agent/agent.py

```diff
@@ -589,6 +589,21 @@ def _check_run_finished(
         try:
             run = job_tracker.run
             status = run.get_status().state
+
+            # Ensure run was not stopped intentionally
+            if status == "preempted":
+                try:
+                    run_state = RunState(
+                        self._api.get_run_state(
+                            self._entity, job_tracker.project, job_tracker.run_id
```

Comment:
We should use the job_tracker entity here. Also we should only do this behavior if the job_tracker entity matches the agent's entity. If the entities dont match this will fail. We should never even attempt to requeue in that case